### ISSUE TITLE
Working with Mercurial topic branches

### ISSUE THREAD

**Issue by johbo** [felipec/git-remote-hg#96]
I am trying to interact with topic branches from Mercurial repositories and I was not yet able to find out if they are supported and how to interact with them.

Background / Context:
* I am trying to interact with Mercurial repositories on heptapod, the merge request workflow is implemented based on topic branches.
* Details - https://heptapod.net/pages/quick-start-guide

What I have observed is that I see the topic branches in the clone which I made via `git`. I was not yet able to push to a new topic branch from `git` though.

Is this already a supported feature or did anyone already figure out a way to successfully interact with those branches from `git` via `git-remote-hg`?

**Comment by felipec** [comment]
I started working on support for topic branches a while ago, but I haven't worked on git-remote-hg recently.

I've pushed the `topic` branch, you can give it a try and see if it works correctly for you and what features are missing: https://github.com/felipec/git-remote-hg/tree/topic

It seems to be a really useful feature that I would like to get into the master branch as soon as possible.

**Comment by felipec** [comment]
@johbo did you try that branch?

**Comment by cmarmo** [comment]
Hi @felipec, I am really interested in having `git-remote-hg` working with mercurial topics.
I have tried your `topic` branch and I didn't succeed in pushing my git branch as a mercurial topic.
Here the error thrown:
```
$ git push origin test-hg-git 
/.../bin/git-remote-hg:79: SyntaxWarning: invalid escape sequence '\w'
  RAW_AUTHOR_RE = re.compile('^(\w+) (?:(.+)? )?<(.*)> (\d+) ([+-]\d+)')
/.../bin/git-remote-hg:261: SyntaxWarning: invalid escape sequence '\('
  m = re.match('^(.+?) ext:\((.+)\)$', name)
searching for changes
no changes found
searching for changes
remote: abort: incompatible Mercurial client; bundle2 required
remote: (see https://www.mercurial-scm.org/wiki/IncompatibleClient)
Everything up-to-date
``` 
I am checking the regexp myself... any help is much appreciated... 🙂 ... thanks!

**Comment by cmarmo** [comment]
After merging `master` in `topic`  the error about regexp is gone but
```
remote: abort: incompatible Mercurial client; bundle2 required
remote: (see https://www.mercurial-scm.org/wiki/IncompatibleClient)
```
is still there.

**Comment by cmarmo** [comment]
For completeness I am using
```
$ hg --version
Mercurial Distributed SCM (version 7.0)
(see https://mercurial-scm.org for more information)
```

**Comment by cmarmo** [comment]
Sorry for bothering again, just to clarify that I have tried with ssh and https protocols and the same error is thrown.
I have also updated mercurial to 7.1
Thanks for listening! 🙂

**Comment by felipec** [comment]
@cmarmo I do have a patch for bundle2 support. I have pushed it to the topic branch, which I have rebased as well.

**Comment by cmarmo** [comment]
Wonderful! Thank you! 🙂 
I've tested the `topic` branch and it looks like it's working now.
However I'm having an error related to my ignorance of Mercurial I guess.
```
mercurial.bundle2.AbortFromPart: rejecting multiple heads on branch "default"
```
I am trying to push something I have created with `git branch` called `test-for-git` : should I call the branch in a "more mercurial" way?
JJust to be clear here are my branches
```bash
$ git branch
  branches/default/tip
* test-for-git
``` 
I am happy to contribute to the documentation once I have succeeded in pushing :)

**Comment by felipec** [comment]
I thought it would be clear if you clone a project with topic branches.

If you want to push your "test-for-git" branch as a Mercurial topic, you need to name it "topics/test-for-git".

**Comment by cmarmo** [comment]
Yep! Thanks! I have renamed the branch and successfully pushed it to the mercurial repo!
Thank you so much!
